application-ownership-models/README.md

```diff
@@ -0,0 +1,116 @@
+
+There is no silver bullet for all upstream applications, so here we need clear and robust, but also flexible ownership model.
+
+Before trying to come up with some definition, let's first check few examples.
+
+#### 2.2.1. Examples of shared applications
+
+While there are more examples, let's take the following two that should paint the picture and describe the issue of shared ownership clearly enough.
+
+**Cluster API provider-independent controllers**
+
+There are three CAPI controllers that are equally used by KaaS teams (Rocket, Hydra, Clippy) - core, kubeadm control plane and kubeadm bootstrap controller. They are all packaged together in the cluster-api-app. All KaaS teams have equal stakes in this app and equal responsibilities.
+
+Which teams is responsible for maintaining the application repository, improving it, automating it, taking care of renovate and other automated pull requests? Which teams takes care of upgrading the app to the latest upstream? Every team for themselves when they have a need? This is not working very efficiently.
+
+**Kubernetes components**
+
+We could say that deploying and using Kubernetes is sort of what we all do for living, right?
+
+Quick reminder that here we are talking about partial application lifecycle ownership, so release engineering and operations, where we can't really say that we do former one, so the latter one is interesting here.
+
+All teams write or package operators that get scheduled to some node, operators that use and maybe sometimes hammer API server, deploy few or few hundreds of CRDs that case clusters to be overloaded, etc.
+
+Let's say that all teams are more or less similarly using all (or most of) Kubernetes components. In that case a single team owning operations for those components would be ~~ludacris~~ ludicrous, and that didn't really work nicely when we tried it.
+
+#### 2.2.2. Implementing shared application ownership in practice
+
+So how do we implement shared application ownership in practice?
+
+Teams own apps with sole application ownership model. And multiple teams take turns, i.e. multiple teams participate in the rotation of sole application ownership.
+
+When a team is assigned to be sole app owner that means that the team takes care of app's release engineering and operations (on-call, postmortems, etc).
+
+How frequently the ownership is rotated depends on the app, and is decided on a case by case basis by the interested teams, but in most cases the app ownership should be passed to the next team every few months to half a year, depending on the release cadence of the app. Less then a month is probably too often, more than half a year could lead to folks falling out of practice (important for operations) or even mean that some team doesn't get to own the app as team structure changes.
```

Comment:
```suggestion
How frequently the ownership is rotated depends on the app, and is decided on a case by case basis by the interested teams, but in most cases the app ownership should be passed to the next team every few months to half a year, depending on the release cadence of the app. Less than a month is probably too often, more than half a year could lead to folks falling out of practice (important for operations) or even mean that some team doesn't get to own the app as team structure changes.
```

Reply:
Though with KaaS restructuration we will solve some problems of ownership we may have shared application that is not owned by a single team. You suggestion could work in that case but we will need to go case by case